Based on https://github.com/joyent/illumos-extra/blob/master/openssh/Patches/0020-Compatibility-fix-for-ListenAddress.patch
From 931242f7495775384b7ce9109b24420d700ac688 Mon Sep 17 00:00:00 2001
From: Alex Wilson <alex.wilson@joyent.com>
Date: Mon, 3 Aug 2015 17:27:41 -0700
Subject: [PATCH 20/30] Compatibility fix for "ListenAddress ::"

In SunSSH, a config that specifies only "ListenAddress ::" in
fact will listen on both IPv4 and IPv6.
---
 servconf.c | 15 +++++++++++++--
 1 file changed, 13 insertions(+), 2 deletions(-)

diff --git a/servconf.c b/servconf.c
--- openssh-6.8p1/servconf.c.~2~	2015-11-06 14:07:17.368752173 +0300
+++ openssh-6.8p1/servconf.c	2015-11-06 14:14:50.139673048 +0300
@@ -1005,6 +1005,15 @@
 		if (arg == NULL || *arg == '\0')
 			fatal("%s line %d: missing address",
 			    filename, linenum);
+                /*
+                * To retain backwards compat with SunSSH, convert any listener
+                * on :: into a listener of unspecified AF (so it listens on
+                * v4 too)
+                */
+		if (strcmp(arg, "::") == 0) {
+			add_listen_addr(options, NULL, 0);
+			break;
+		}
 		/* check for bare IPv6 address: no "[]" and 2 or more ":" */
 		if (strchr(arg, '[') == NULL && (p = strchr(arg, ':')) != NULL
 		    && strchr(p+1, ':') != NULL) {
